src/codemodder/dependency_management/setupcfg_writer.py

```diff
@@ -0,0 +1,124 @@
+import configparser
+
+from typing import Optional
+from codemodder.dependency import Dependency
+from codemodder.change import ChangeSet
+from codemodder.dependency_management.base_dependency_writer import DependencyWriter
+from codemodder.diff import create_diff_and_linenums
+from codemodder.logging import logger
+
+
+import re
+
+
+def find_leading_whitespace(s):
+    match = re.match(r"(\s+)", s)
+    if match:
+        return match.group(1)
+    return ""
+
+
+def added_line_nums_strategy(lines, i):
+    return lines[i]
+
+
+class SetupCfgWriter(DependencyWriter):
+    def add_to_file(
+        self, dependencies: list[Dependency], dry_run: bool = False
+    ) -> Optional[ChangeSet]:
+        config = configparser.ConfigParser()
+
+        try:
+            config.read(self.path)
+        except configparser.ParsingError:
+            logger.debug("Unable to parse setup.cfg file.")
+            return None
+
+        if "options" not in config or not (
+            defined_dependencies := config["options"].get("install_requires", "")
+        ):
+            logger.debug("Unable to add dependencies to setup.cfg file.")
+            return None
+
+        with open(self.path, "r", encoding="utf-8") as f:
+            original_lines = f.readlines()
+
+        new_lines = self.build_new_lines(
+            original_lines, defined_dependencies, dependencies
+        )
+        if not new_lines:
+            logger.debug("Unable to add dependencies to setup.cfg file.")
+            return None
+
+        if not dry_run:
+            try:
+                with open(self.path, "w", encoding="utf-8") as f:
+                    f.writelines(new_lines)
+            except Exception:
+                logger.debug("Unable to add dependencies to setup.cfg file.")
+                return None
+
+        diff, added_line_nums = create_diff_and_linenums(original_lines, new_lines)
+
+        changes = self.build_changes(
+            dependencies, added_line_nums_strategy, added_line_nums
+        )
+        return ChangeSet(
+            str(self.path.relative_to(self.parent_directory)),
+            diff,
+            changes=changes,
+        )
+
+    def build_new_lines(
+        self,
+        original_lines: list[str],
+        defined_dependencies: str,
+        dependencies_to_add: list[Dependency],
+    ) -> Optional[list[str]]:
+        """
+        configparser does not retain formatting or comment lines, so we have to build
+        the output newline manually.
+        """
+        clean_lines = list(map(lambda s: s.strip(), original_lines))
```

Comment:
I like `map` but genuinely curious whether there's any benefit to doing it this way rather than using a comprehension?
```python
clean_lines = [s.strip for s in original_lines]
```

Reply:
no other benefit to map, I think the moment I wrote it that's the first thing that came to me. I'll switch it out